secant/Features/App/App.swift

```diff
@@ -269,7 +269,12 @@ extension AppReducer {
             state.route = .phraseDisplay
 
         case .phraseDisplay(.finishedPressed):
-            state.route = .phraseValidation
+            if let storedWallet = try? environment.walletStorage.exportWallet(),
+               storedWallet.hasUserPassedPhraseBackupTest {
+                state.route = .home
+            } else {
+                state.route = .phraseValidation
+            }
```

Comment:
This is actually very interesting problem and a good catch. The solution here "works" and will very likely work for some time. You mentioned that "show me display again" can be accessed from profile screen. The code

```swift
if let storedWallet = try? environment.walletStorage.exportWallet(),
               storedWallet.hasUserPassedPhraseBackupTest {
                state.route = .home
            }
```
always returns user back to home as long as test has been passed. What if we want to go back to profile screen? Or any other screen. This opens a discussion how to handle MANY options in the TCA routing. The action
```swift
case .phraseDisplay(.finishedPressed):
```
is triggered when the button is pressed but it's not helping to determine "the next" route. I'm thinking we need some better way how to be able to decide the next route when this finished button is pressed. We very likely need to know the previous route so we can get some context.

What do you say if we reimplement the route so it is computed property and whenever newValue is set, we backup the previous one. Something like this:
```swift
    var prevRoute: Route?
    var internalRoute: Route = .welcome
    var route: Route {
        get { internalRoute }
        set {
            prevRoute = internalRoute
            internalRoute = newValue
        }
    }
```
All will work as is but we now have the information about the previous route, meaning we can react in the 
```swift
case .phraseDisplay(.finishedPressed):
```
If prev was home, let's get back to home. If it was profile, let's get back to profile. If it was phraseValidation, it means user wanted to see the phrase again. If it was onboarding/create wallet, we need to trigger the test.

If you think the solution in this PR is good enough, consider this comment pointless and task is good to merge.